configure.ac

```diff
@@ -305,6 +263,79 @@ else
   esac
 fi
 
+# This should come after all tests which modify CFLAGS and friends.
+if test x"$use_ecmult_static_precomputation" != x"no"; then
+  if test x"$cross_compiling" = x"no"; then
+    set_precomp=yes
+    if test x"${CC_FOR_BUILD+x}${CFLAGS_FOR_BUILD+x}${CPPFLAGS_FOR_BUILD+x}${LDFLAGS_FOR_BUILD+x}" != x; then
+      AC_MSG_WARN([CC_FOR_BUILD, CFLAGS_FOR_BUILD, CPPFLAGS_FOR_BUILD, and/or LDFLAGS_FOR_BUILD is set but ignored because we are not cross-compiling.])
+    fi
+    # If we're not cross-compiling, simply use the same compiler for building the static precompation code.
+    CC_FOR_BUILD="$CC"
+    CFLAGS_FOR_BUILD="$CFLAGS"
+    CPPFLAGS_FOR_BUILD="$CPPFLAGS"
+    LDFLAGS_FOR_BUILD="$LDFLAGS"
+  else
```

Comment:
@sipa I think this is one part. But yes, I can split it if you want. I was somewhat lazy because I was hacking on the code at various places and then just had a bunch of changes...

Reply:
@real-or-random Yes, sorry, I realize that that is what this part does. The code looks good, but it's easier to ascertain it doesn't introduce other changes if the move-only part and the rest are in separate commits.